codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/CustomizableOperationGenerator.kt

```diff
@@ -202,6 +204,13 @@ class CustomizableOperationGenerator(
                             self
                         }
 
+                        /// Adds a runtime plugin.
+                        ##[doc(hidden)]
+                        pub fn runtime_plugin(mut self, runtime_plugin: impl #{RuntimePlugin} + 'static) -> Self {
```

Comment:
does this really need to be pub? I kinda imagine pub(crate) would suffice. I also might name it `with_plugin(...)` to clarify

Reply:
I kind of broke our builder convention with the `with_` methods, but I figured that's OK since it's for something more internal rather than the frequently used public facing API. Now that a lot of our public builders have getters though, we may consider bringing in the `with_` convention rather than having `thing`, `set_thing`, and `get_thing`.

Reply:
Made `pub(crate)` in cacf7bac9993f8785ba087aa185e4fb69d5c1419